chanfitness/revenue.go

```diff
@@ -0,0 +1,52 @@
+package chanfitness
+
+import (
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// attributeIncoming is the percentage of fees that are attributed to the
+// incoming channel in a forwarding event. The remaining 1- attributeIncoming
+// percent is attributed to the outgoing channel.
+var attributeIncoming = 0.5
```

Comment:
One lower level detail that's relevant here is that the fee policy if the _outgoing_ channel is the one that's applied during forwarding/settlement. We accept the incoming HTLC, but then our bandwidth is allocated on the outgoing HTLC, which is where we actually charge our fees: we send out less than we send in. However, the _incoming_ channel is actually where the funds are accrued since there's where the difference between the incoming+outgoing HTLC is stored.

Reply:
One other direction that may be worth exploring is exactly how we surface this information on the RPC interface. Rather than showing a single value (total revenue) for a channel, we could instead unpack the revenue into channel pairs. This would allow the end caller to reason about the relationship between the target channel and the set of open channels, essentially: typically where is the incoming HTLC being sourced from?